x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/rules_list_column_selector.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiDataGridColumn,
+  EuiTableActionsColumnType,
+  EuiTableComputedColumnType,
+  EuiTableFieldDataColumnType,
+  useDataGridColumnSelector,
+} from '@elastic/eui';
+import React, { useMemo, useState } from 'react';
+import { useKibana } from '../../../../common/lib/kibana';
+import { RuleTableItem } from '../../../../types';
+
+type RulesListTableColumns =
+  | EuiTableFieldDataColumnType<RuleTableItem>
+  | EuiTableComputedColumnType<RuleTableItem>
+  | EuiTableActionsColumnType<RuleTableItem>;
+
+export type RulesListColumns = {
+  id?: RulesListVisibleColumns;
+  selectorName?: string;
```

Comment:
nitpick: this seems a bit odd since we're passing in this modified type extended with `selectorName` but returning the base RulesListTableColumns, I guess technically they're compatible but maybe we should just always use this modified type? Would that break the EUI table columns prop?

Reply:
yes, it will break the EUI table columns prop, that's why I did it like that